### PR TITLE
Use static instance of Aqua TypeResolver to avoid memory leak in EF Core execution plan cache

### PR DESCRIPTION
**Pull request by vluedeling**

### CODE REVIEW

**Comment by vluedeling** [comment]
See Issue #3

**Comment by vluedeling** [comment]
Hello Alexander,
we have identified a memory leak in our deployment of InfoCarrier Core. We have a WPF client which sends queries to the InfoCarrier server, which in turn queries an SQL server database through Entity Framework Core. We use your InfoCarrier Core framework in version 2.2.6.

We have, since recently, a query which is executed every 5 seconds. In the server, we notice a steady increase in memory consumption whenever clients are connected.

We have narrowed the problem down to the usage of anonymous types ("new { ... }") in the query, like so:

```
                    var maxTimes = dbContext.Set<EventCapture>()
                        .GroupBy(
                            x => x.MachineId,
                            (x, y) => new { MachineId = x, MaxTime = y.Max(a => a.Timestamp) })
                        .Select(x => new EventCapture() { MachineId = x.MachineId, Timestamp = x.MaxTime })
                        .ToList();

```

The profiler shows that Entity Framework Core creates new Cache Entries for the execution plans in every request, which is only supposed to happen once per distinct statement. Those plans remain in the cache and are never purged.

Your QueryDataHelper instantiates a new TypeResolver on every request, which causes the Aqua TypeSystem to always create new Runtime Types, which in turn make EF Core think that it has encountered a new type of query.

We could solve the issue with a rather straightforward change. Instead of having the QueryDataHelper create a new TypeResolver on every request, we use the static instance field provided by the TypeResolver. This makes the memory leak disappear (and also increases performance for client requests).

I suspect that this problem will also arise in the current version of InfoCarrier Core, because it also uses new TypeResolver instances for requests.

I will upload our my proposed change as a pull request.

Regards,

Volker

File: src/InfoCarrier.Core/Server/QueryDataHelper.cs [hunk 1]
```diff
@@ -39,7 +39,6 @@ internal class QueryDataHelper
         private readonly DbContext dbContext;
         private readonly IEnumerable<IInfoCarrierValueMapper> valueMappers;
         private readonly System.Linq.Expressions.Expression linqExpression;
-        private readonly ITypeResolver typeResolver = new TypeResolver();
```
Comment (azabluda) [inline src/InfoCarrier.Core/Server/QueryDataHelper.cs:42]: To make the forward merges to 3.x and 5.x branches smoother, please fix QueryDataHelper the same way as InfoCarrierQueryCompiler, e.g. keep the field but just initialize it with the static instance.

**Review by azabluda — CHANGES_REQUESTED** [review]
Hi Volker, thanks for your contribution. The suggested fix generally looks good, but please make it a bit less intrusive (see comment) to facilitate further merges. Thank you, Alexander

File: src/InfoCarrier.Core/Server/QueryDataHelper.cs (hunk 1 above)
Comment (vluedeling) [reply src/InfoCarrier.Core/Server/QueryDataHelper.cs:42]: I have reintroduced the typeResolver field, you should have less work with the upstresm merge now.

**Review by azabluda — APPROVED** [review]
👍

**Pull request merged**